core/src/test/java/org/apache/iceberg/TestFastAppend.java

```diff
@@ -478,4 +478,13 @@ public void testIncludedPartitionSummaryLimit() {
     String changedPartitions = table.currentSnapshot().summary().get(SnapshotSummary.CHANGED_PARTITION_COUNT_PROP);
     Assert.assertEquals("Should set changed partition count", "2", changedPartitions);
   }
+
+  @Test(expected = UnsupportedOperationException.class)
+  public void testAppendToBranch() throws UnsupportedOperationException {
+    table.newFastAppend()
+            .appendFile(FILE_A)
+            .commit();
+    table.manageSnapshots().createBranch("ref", table.currentSnapshot().snapshotId()).commit();
+    table.newFastAppend().appendFile(FILE_B).toBranch("ref").commit();
```

Comment:
Could this implement branch commits for `FastAppend`? That way we can see one operation that works from end-to-end when reviewing the changes to `SnapshotUpdate`.

Reply:
@rdblue Have written the complete test-case in https://github.com/apache/iceberg/pull/5010.